pkg/txgenerator/erc1155.go

```diff
@@ -0,0 +1,161 @@
+package txgenerator
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/status-im/go-wallet-sdk/pkg/contracts/erc1155"
+)
+
+// TransferERC1155Params contains parameters for generating an ERC1155 single token transfer transaction
+type TransferERC1155Params struct {
+	BaseTxParams
+	// TokenAddress is the ERC1155 token contract address
+	TokenAddress common.Address
+	// From is the current owner address
+	From common.Address
+	// To is the recipient address
+	To common.Address
+	// TokenID is the token ID to transfer
+	TokenID *big.Int
+	// Value is the amount of tokens to transfer
+	Value *big.Int
+}
+
+// BatchTransferERC1155Params contains parameters for generating an ERC1155 batch token transfer transaction
+type BatchTransferERC1155Params struct {
+	BaseTxParams
+	// TokenAddress is the ERC1155 token contract address
+	TokenAddress common.Address
+	// From is the current owner address
+	From common.Address
+	// To is the recipient address
+	To common.Address
+	// TokenIDs is the array of token IDs to transfer
+	TokenIDs []*big.Int
+	// Values is the array of amounts to transfer (corresponding to TokenIDs)
+	Values []*big.Int
+}
+
+// SetApprovalForAllERC1155Params contains parameters for generating an ERC1155 setApprovalForAll transaction
+type SetApprovalForAllERC1155Params struct {
+	BaseTxParams
+	// TokenAddress is the ERC1155 token contract address
+	TokenAddress common.Address
+	// Operator is the operator address to approve or revoke
+	Operator common.Address
+	// Approved indicates whether to approve (true) or revoke (false) the operator
+	Approved bool
+}
+
+// TransferERC1155 generates an unsigned transaction for transferring a single ERC1155 token
+// The transaction type is determined by the presence of GasPrice (legacy) or MaxFeePerGas (EIP-1559)
+func TransferERC1155(params TransferERC1155Params) (*types.Transaction, error) {
+	if params.TokenAddress == (common.Address{}) {
+		return nil, errors.New("token address cannot be zero")
+	}
+	if params.From == (common.Address{}) {
+		return nil, errors.New("from address cannot be zero")
+	}
+	if params.TokenID == nil || params.TokenID.Sign() < 0 {
+		return nil, errors.New("token ID must be non-negative")
+	}
+	if params.Value == nil || params.Value.Sign() < 0 {
+		return nil, errors.New("value must be non-negative")
+	}
+	if params.ChainID == nil {
```

Comment:
Then here and in other places below we could do:
```go
    txType, err := validateAndDetermineTxType(params.BaseTxParams)
	if err != nil {
		return nil, err
	}
```

and do the calls below only if params are valid and pass `txType` to `createTokenTransaction` call.